tests/phpunit/test-class-wp-widget-image.php

```diff
@@ -0,0 +1,230 @@
+
+		// Should filter invalid image alignment.
+		$result = $widget->update( array( 'align' => 'next to caption' ), $instance );
+		$this->assertSame( $result, $instance );
+
+		// Should return valid image caption.
+		$expected = array( 'caption' => 'A caption with <a href="#">link</a>' );
+		$result = $widget->update( $expected, $instance );
+		$this->assertSame( $result, $expected );
+
+		// Should filter invalid image caption.
+		$result = $widget->update( array( 'caption' => '"><i onload="alert(\'hello\')" />' ), $instance );
+		$this->assertSame( $result, $instance );
+
+		// Should return valid alt text.
+		$expected = array( 'alt' => 'A water tower' );
+		$result = $widget->update( $expected, $instance );
+		$this->assertSame( $result, $expected );
+
+		// Should filter invalid alt text.
+		$result = $widget->update( array( 'alt' => '"><i onload="alert(\'hello\')" />' ), $instance );
+		$this->assertSame( $result, array( 'alt' => '">' ) );
+
+		// Should return valid link type.
+		$expected = array( 'link_type' => 'file' );
+		$result = $widget->update( $expected, $instance );
+		$this->assertSame( $result, $expected );
+
+		// Should filter invalid link type.
+		$result = $widget->update( array( 'link_type' => 'interesting' ), $instance );
+		$this->assertSame( $result, $instance );
+
+		// Should return valid link url.
+		$expected = array( 'link_url' => 'https://example.org' );
+		$result = $widget->update( $expected, $instance );
+		$this->assertSame( $result, $expected );
+
+		// Should filter invalid link url.
+		$result = $widget->update( array( 'link_url' => 'not_a_url' ), $instance );
+		$this->assertNotSame( $result, $instance );
+		$this->assertStringStartsWith( 'http://', $result['link_url'] );
+
+		// Should return valid image classes.
+		$expected = array( 'image_classes' => 'A water tower' );
+		$result = $widget->update( $expected, $instance );
+		$this->assertSame( $result, $expected );
+
+		// Should filter invalid image classes.
+		$result = $widget->update( array( 'image_classes' => '"><i onload="alert(\'hello\')" />' ), $instance );
+		$this->assertSame( $result, array( 'image_classes' => '">' ) );
+
+		// Should return valid link classes.
+		$expected = array( 'link_classes' => 'A water tower' );
+		$result = $widget->update( $expected, $instance );
+		$this->assertSame( $result, $expected );
+
+		// Should filter invalid link classes.
+		$result = $widget->update( array( 'link_classes' => '"><i onload="alert(\'hello\')" />' ), $instance );
+		$this->assertSame( $result, array( 'link_classes' => '">' ) );
+
+		// Should return valid rel text.
+		$expected = array( 'link_rel' => 'previous' );
+		$result = $widget->update( $expected, $instance );
+		$this->assertSame( $result, $expected );
+
+		// Should filter invalid rel text.
+		$result = $widget->update( array( 'link_rel' => '"><i onload="alert(\'hello\')" />' ), $instance );
+		$this->assertSame( $result, array( 'link_rel' => '">' ) );
```

Comment:
I don't know, I couldn't think of a better way to test `sanitize_text_field` off hand. Any idea for a better one?